pkg/kubelet/kubelet.go

```diff
@@ -387,6 +387,12 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) && cloudprovider.IsDeprecatedInternal(cloudProvider) {
+		return nil, fmt.Errorf(
+			"cloud provider %q was specified, but built-in cloud providers are disabled. Please set --cloud-provider=external and migrate to an external cloud provider",
```

Comment:
Might be good to mention the possible need for a credential provider and suggest reaching out to SIG Cloud Provider.

Reply:
Added this line to address the request: https://github.com/kubernetes/kubernetes/pull/100136/files#diff-67dd9e8f3ee257072765326cb4f242852554a2c0753563fa51e292c0a63a7b94R394